### PR TITLE
ra-multiplex: init at 0.2.2

### PR DESCRIPTION
**Pull request by NorfairKing**
###### Description of changes

A packaging of https://github.com/pr2502/ra-multiplex

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by tomodachi94 — APPROVED** [review]
Lgtm.

File: pkgs/development/tools/rust/ra-multiplex/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+rustPlatform.buildRustPackage {
+  pname = "ra-multiplex";
+  version = "0.2.2";
+  cargoSha256 = "sha256-Z0ul50LeSb3qhdfuvRYvNAExtfH8y2+IkKLtqtlaL9E=";
+
+  src = fetchFromGitHub {
+    owner = "pr2502";
+    repo = "ra-multiplex";
+    rev = "dcb5f83890cb91016b0a1590cc1b732606bb6ec1";
+    sha256 = "sha256-Hf4Gj9eXEP4gXiqNV4Jq0oiGLX3DtDF9At1feEZ+bUE=";
```
Comment (oxalica) [inline pkgs/development/tools/rust/ra-multiplex/default.nix:14]: No need to dup hash algorithm names.
```suggestion
    hash = "sha256-Hf4Gj9eXEP4gXiqNV4Jq0oiGLX3DtDF9At1feEZ+bUE=";
```

File: pkgs/development/tools/rust/ra-multiplex/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  };
+
+  # We make two extra scripts:
```
Comment (oxalica) [inline pkgs/development/tools/rust/ra-multiplex/default.nix:17]: Where are these scripts from? They do not seem to be required, and are not mentioned upstream. If they are just to make *YOU* easier to use, they should go to your personal profile.

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -16737,6 +16737,7 @@ with pkgs;
     inherit (darwin.apple_sdk.frameworks) CoreServices;
   };
   rust-analyzer = callPackage ../development/tools/rust/rust-analyzer/wrapper.nix { };
+  ra-multiplex = callPackage ../development/tools/rust/ra-multiplex/default.nix {};
```
Comment (oxalica) [inline pkgs/top-level/all-packages.nix:16740]: Please sorted it alphabetically.

File: pkgs/development/tools/rust/ra-multiplex/default.nix [hunk 4]
```diff
@@ -0,0 +1,47 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+}:
+rustPlatform.buildRustPackage {
+  pname = "ra-multiplex";
+  version = "0.2.2";
+  cargoSha256 = "sha256-Z0ul50LeSb3qhdfuvRYvNAExtfH8y2+IkKLtqtlaL9E=";
```
Comment (oxalica) [inline pkgs/development/tools/rust/ra-multiplex/default.nix:8]:
```suggestion
  cargoHash = "sha256-Z0ul50LeSb3qhdfuvRYvNAExtfH8y2+IkKLtqtlaL9E=";
```

**Review by oxalica — CHANGES_REQUESTED** [review]

**Review by oxalica — APPROVED** [review]
> It's not clear in which range you want me to sort this. I put it next to rust-analyzer before, so now I put it above rustfmt.

I meant in the part where most of them are rust tools. I'd put it between `ravedude` and `rhack`. But the list is not strictly sorted in the first place, so it does not matter much.

Besides this, diff LGTM.

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -16730,6 +16730,8 @@ with pkgs;
 
   ravedude = callPackage ../development/tools/rust/ravedude { };
 
+  ra-multiplex = callPackage ../development/tools/rust/ra-multiplex/default.nix {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:16733]:
```suggestion
  ra-multiplex = callPackage ../development/tools/rust/ra-multiplex {};
```

File: pkgs/development/tools/rust/ra-multiplex/default.nix [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  pname = "ra-multiplex";
+  version = "0.2.2";
+  cargoHash = "sha256-MeUtkPjOsL1kQ2W0Q1/OqhKDVXs4cECkATHISpyfp9U=";
+
+  src = fetchFromGitHub {
+    owner = "pr2502";
+    repo = "ra-multiplex";
+    rev = "dcb5f83890cb91016b0a1590cc1b732606bb6ec1";
+    hash = "sha256-Hf4Gj9eXEP4gXiqNV4Jq0oiGLX3DtDF9At1feEZ+bUE=";
+  };
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/ra-multiplex/default.nix:15]:
```suggestion
}:

rustPlatform.buildRustPackage {
  pname = "ra-multiplex";
  version = "0.2.2";

  src = fetchFromGitHub {
    owner = "pr2502";
    repo = "ra-multiplex";
    rev = "dcb5f83890cb91016b0a1590cc1b732606bb6ec1";
    hash = "sha256-Hf4Gj9eXEP4gXiqNV4Jq0oiGLX3DtDF9At1feEZ+bUE=";
  };

  cargoHash = "sha256-MeUtkPjOsL1kQ2W0Q1/OqhKDVXs4cECkATHISpyfp9U=";
```

**Review by SuperSandro2000 — COMMENTED** [review]
not sure if we should wrap rust-analyzer

**Comment by SuperSandro2000** [comment]
> > not sure if we should wrap rust-analyzer
> 
> I don't understand what you mean by this. Could you elaborate?

The tool requires rust-analyzer to be present in PATH https://github.com/pr2502/ra-multiplex#how-to-use

We can use wrapProgram to make sure that is always present in PATH and people don't need to configure that. see https://github.com/NixOS/nixpkgs/blob/684c17c429c42515bafb3ad775d2a710947f3d67/doc/languages-frameworks/ruby.section.md#packaging-executables-that-require-wrapping-packaging-executables-that-require-wrapping for an example

**Comment by SuperSandro2000** [comment]
> We can use wrapProgram to make sure that is always present in PATH and people don't need to configure that. see

I have ammended that

**Pull request merged**